### PR TITLE
[WFLY-17041] Eliminating usage of org.jboss.msc.value.Value interface in ViewDescription

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFLY-17041

### CODE REVIEW

File: ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java [hunk 1]
```diff
@@ -179,12 +179,25 @@ public Deque<ViewConfigurator> getConfigurators() {
...
      *
      * @param serviceName     The view service name
-     * @param viewClassLoader
+     * @param viewClassLoader The view class loader
+     * @param appclient       appclient environment
+     * @deprecated use {@link ViewDescription#createInjectionSource(ServiceName, Supplier, boolean)} instead. This method will be removed in the future.
      */
     protected InjectionSource createInjectionSource(final ServiceName serviceName, Value<ClassLoader> viewClassLoader, boolean appclient) {
         return new ViewBindingInjectionSource(serviceName);
     }
```
Comment (chengfang) [inline ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java:188]: any reason this method is not removed, or the Value not converted to Supplier? It's overloaded with the next one. A quick check show no usage of this protected method.

File: ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java [hunk 2]
```diff
@@ -179,12 +179,25 @@ public Deque<ViewConfigurator> getConfigurators() {
...
+    /**
+     * Create the injection source
+     *
+     * @param serviceName     The view service name
+     * @param viewClassLoader The view class loader
+     * @param appclient       appclient environment
+     */
+    protected InjectionSource createInjectionSource(final ServiceName serviceName, Supplier<ClassLoader> viewClassLoader, boolean appclient) {
+        return new ViewBindingInjectionSource(serviceName);
+    }
```
Comment (chengfang) [inline ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java:199]: this method only uses the first param and ignoring the other 2 params. Can we just remove the 2 params?

File: ejb3/src/main/java/org/jboss/as/ejb3/component/EJBViewDescription.java [hunk 3]
```diff
@@ -36,7 +36,8 @@
 import org.jboss.invocation.proxy.ProxyFactory;
 import org.jboss.metadata.ejb.spec.MethodInterfaceType;
 import org.jboss.msc.service.ServiceName;
-import org.jboss.msc.value.Value;
+
+import java.util.function.Supplier;
```
Comment (chengfang) [inline ejb3/src/main/java/org/jboss/as/ejb3/component/EJBViewDescription.java:41]: can we move the java.* import to the top, which I think is the convention in WF?

File: ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java (hunk 1 above)
Comment (ropalka) [reply ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java:188]: I removed this BC method. You are right it is not necessary. Only code in WildFly used it.

File: ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java (hunk 2 above)
Comment (ropalka) [reply ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java:199]: We cannot. It is template method overwritten by [EJBViewDescription](https://github.com/wildfly/wildfly/blob/main/ejb3/src/main/java/org/jboss/as/ejb3/component/EJBViewDescription.java#L105).

File: ejb3/src/main/java/org/jboss/as/ejb3/component/EJBViewDescription.java (hunk 3 above)
Comment (ropalka) [reply ejb3/src/main/java/org/jboss/as/ejb3/component/EJBViewDescription.java:41]: Fixed

**Review by chengfang — APPROVED** [review]

**Pull request merged**